woocommerce/woocommerce-gutenberg-products-block
# Add pattern chooser in Product Collection

## What

This PR adds the Pattern Chooser to Product Collection. Patterns that are currently available in Product Collection are purely UI patterns.
That means when you choose some pattern, then layout settings (grid vs list, number of columns, number products per page, pages, offset) are taken from the pattern and they override the current setup. But all the other query attributes are preserved, so changing the pattern doesn't influence the filters that are applied to Product Collection block.

Additionally:
- existing patterns have been enabled in inserter (some of them were hidden)
- `DisplayLayoutToolbar` has been moved from Inspector Controls to Block Controls
- Number of columns in Modal is responsive

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9372

## Why

It allows to change patterns from the Block Controls.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Go to Editor
2. Add Product Collection blocks with different configiurations, e.g.
    - one that inherits query from template
    - one that has "On Sale" filter enabled with limited number of products per page
    - one that shows handpicked products
3. Focus on first Product Collection block you added and click "Choose pattern"
4. Expected: Modal with 4 patterns is displayed
<img width="1559" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/0356072a-5924-4a47-8793-05176478c8d0">
5. Click some pattern 
6. Expected: Layout of Product Collection has changed reflecting what you chose, but the query settings are preserved (e.g. enabled inherit query from template or "On Sale" filter)
7. Repeat for other Product Collection blocks.


## Screenshots or screencast

https://github.com/woocommerce/woocommerce-blocks/assets/20098064/edfd2ddc-4cf1-41aa-a375-bbd5201bb603

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Product Collection: Add possibility to switch patterns
> Product Collection: Enable Product Collection Grid and Product Collection Full Grid patterns
